### PR TITLE
Proper logs and error messages

### PR DESCRIPTION
**Pull request by anjalimukundan**

### CODE REVIEW

File: cypress/support/cypress-commands/assertion.js [hunk 1]
```diff
@@ -639,7 +652,7 @@ Cypress.Commands.add(
 
     fireLog.info('Event Received Check : ' + eventReceivedCheck);
     fireLog.info('Event Schema Check : ' + schemaCheck);
-    fireLog.info('Event Content Check : ' + contentCheck);
+    // fireLog.info('Event Content Check : ' + contentCheck);
```
Comment (jbigel) [inline cypress/support/cypress-commands/assertion.js:655]: If we don't need this code can we remove it?

File: cypress/support/validations/decodeValidation.js [hunk 2]
```diff
@@ -159,7 +163,12 @@ class decodeValidations {
             `Regular Expression Validation: expected ${param} to be in a valid ${regexFormat} format`,
             'decodeBase64AndJwtToken'
           ).then(() => {
-            assert.equal(true, resultSet, 'RegEx Validation:');
+            // assert.equal(true, resultSet, 'RegEx Validation:');
```
Comment (jbigel) [inline cypress/support/validations/decodeValidation.js:166]: If we don't need this code can we remove it?

**Review by jbigel — CHANGES_REQUESTED** [review]

File: cypress/support/cypress-commands/assertion.js (hunk 1 above)
Comment (anjalimukundan) [reply cypress/support/cypress-commands/assertion.js:652]: removed

**Review by neeradanelxsi — APPROVED** [review]

File: cypress/support/cypress-commands/assertion.js [hunk 3]
```diff
@@ -433,25 +451,25 @@ function loggingValidationCheckResult(validationCheck) {
...
-      validationCheck.forEach((assertion) => {
-        if (assertion.validationStatus == CONSTANTS.SKIPPED) {
-          assert.equal(assertion.validationStatus, CONSTANTS.SKIPPED, assertion.validationPoint);
-        } else {
-          assert.equal(assertion.validationStatus, CONSTANTS.PASS, assertion.validationPoint);
-        }
-      });
-    });
+    } else {
+      cy.log(
```
Comment (jbigel) [inline cypress/support/cypress-commands/assertion.js:463]: Can we use `fireLog` for consistency?

File: cypress/support/cypress-commands/assertion.js [hunk 4]
```diff
@@ -433,25 +451,25 @@ function loggingValidationCheckResult(validationCheck) {
...
   // Printing the status of all checks in the report.
-  cy.get(validationCheck)
-    .each((logging) => {
+  cy.get(validationCheck);
+  // Assume the checks. If anything is marked other than skipped or pass, then fail the testcase.
+  validationCheck.forEach((assertion) => {
+    if (assertion.validationStatus == CONSTANTS.SKIPPED) {
       cy.log(
-        `${logging.validationPoint}: ${logging.validationStatus}. ${logging.message}`,
+        `${assertion.validationPoint}: ${assertion.validationStatus}. ${assertion.message}`,
```
Comment (jbigel) [inline cypress/support/cypress-commands/assertion.js:459]: Can we use `fireLog` for consistency?

File: cypress/support/validations/moduleValidations/advertisingMiscValidation.js [hunk 5]
```diff
@@ -51,18 +51,19 @@ function validateAdvertisingAdvertisingId(method, validationTypeObject, apiOrEve
   const pretext = `${CONSTANTS.METHOD_CONTENT} for ${method} method : `;
 
   // Ifa is an advertising identifier and it should not be empty.
-  cy.log(
-    `${pretext} Ifa is not null or undefined: *******`,
-    'validateAdvertisingAdvertisingId'
-  ).then(() => {
-    assert.exists(response.ifa, `${pretext} : Is not null or undefined`);
-  });
+  cy.log(`${pretext} Ifa value is not null or undefined`, 'validateAdvertisingAdvertisingId').then(
```
Comment (jbigel) [inline cypress/support/validations/moduleValidations/advertisingMiscValidation.js:54]: Can we use `fireLog` for consistency?

File: cypress/support/validations/moduleValidations/advertisingMiscValidation.js [hunk 6]
```diff
@@ -51,18 +51,19 @@ function validateAdvertisingAdvertisingId(method, validationTypeObject, apiOrEve
...
+      assert.exists(response.ifa, `${pretext} : Is not null or undefined`);
+    }
+  );
 
   switch (ScenarioType) {
     case CONSTANTS.LIMITADTRACKING_ON:
       // lmt value equal to 1 when limitAdTracking is on
       cy.log(
-        `${pretext} Lmt equal to be 1: ` + response.lmt,
+        `${pretext} Expected Lmt value : ${CONSTANTS.ADVERTISING_LIMITIADTRACKING_ON_LMT},` +
```
Comment (jbigel) [inline cypress/support/validations/moduleValidations/advertisingMiscValidation.js:64]: Can we use `fireLog` for consistency?

File: cypress/support/validations/moduleValidations/deviceMiscValidation.js [hunk 7]
```diff
@@ -39,9 +39,11 @@ export function deviceMiscValidation(method, validationTypeObject, apiOrEventObj
     case CONSTANTS.DEVICE_HDCP:
     case CONSTANTS.DEVICE_HDR:
       for (const key in response) {
-        cy.log(`Expected ${key} value to be boolean`).then(() => {
-          assert.isBoolean(response[key], 'Expected to be');
-        });
+        cy.log(`Miscellaneous Validation for ${method}: Expected ${key} value to be boolean`).then(
```
Comment (jbigel) [inline cypress/support/validations/moduleValidations/deviceMiscValidation.js:42]: Can we use `fireLog` for consistency?

**Review by jbigel — CHANGES_REQUESTED** [review]

File: cypress/support/cypress-commands/assertion.js (hunk 4 above)
Comment (anjalimukundan) [reply cypress/support/cypress-commands/assertion.js:459]: updated

File: cypress/support/cypress-commands/assertion.js (hunk 3 above)
Comment (anjalimukundan) [reply cypress/support/cypress-commands/assertion.js:463]: updated

File: cypress/support/validations/moduleValidations/advertisingMiscValidation.js (hunk 5 above)
Comment (anjalimukundan) [reply cypress/support/validations/moduleValidations/advertisingMiscValidation.js:54]: updated

File: cypress/support/validations/moduleValidations/advertisingMiscValidation.js (hunk 6 above)
Comment (anjalimukundan) [reply cypress/support/validations/moduleValidations/advertisingMiscValidation.js:64]: updated to fireLog

File: cypress/support/validations/moduleValidations/deviceMiscValidation.js (hunk 7 above)
Comment (anjalimukundan) [reply cypress/support/validations/moduleValidations/deviceMiscValidation.js:42]: updated to fireLog

File: cypress/TestCases/FireboltCertification/Advertising.feature [hunk 8]
```diff
@@ -141,10 +141,10 @@ Feature: Advertising
...
 
-   @sdk @transport @requiresPlatformImplementation
-   Scenario: Advertising.onPolicyChanged - Positive Scenario: Clearing event listeners
-      When '3rd party app' registers for the 'advertising onPolicyChanged' event using the 'Firebolt' API
-      And I clear 'advertising onPolicyChanged event' listeners
-      And 1st party app invokes the 'Firebolt' API to 'set true for allowAppContentAdTargeting'
-      Then 'Firebolt' platform responds to '1st party app' with 'set true for allowAppContentAdTargeting'
-      And 'Firebolt' platform does not trigger event for 'onAdvertisingPolicyChanged'
\ No newline at end of file
+      @Advertising @coreSDK @sdk @transport @requiresPlatformImplementation
```
Comment (neeradanelxsi) [inline cypress/TestCases/FireboltCertification/Advertising.feature:144]: check the latest dev, tags are not correct

**Review by jbigel — APPROVED** [review]

**Review by neeradanelxsi — APPROVED** [review]

File: cypress/TestCases/FireboltCertification/Advertising.feature (hunk 8 above)
Comment (anjalimukundan) [reply cypress/TestCases/FireboltCertification/Advertising.feature]: updated the tags

**Review by ibrahimshuail — APPROVED** [review]

**Review by Abhishk123 — APPROVED** [review]

**Pull request merged**